tests/lib/linter/linter.js

```diff
@@ -4779,6 +4819,59 @@ var a = "test2";
         });
     });
 
+    describe("warnInlineConfig option", () => {
```

Comment:
There's no `warnInlineConfig` linter option. It's calculated internally in [`normalizeVerifyOptions`](https://github.com/eslint/eslint/blob/55c1685aab0fa75074ac1033abfa6e34cb7baaf6/lib/linter/linter.js#L617).

This test is passing because of `noInlineConfig: true` in the config, not because of `warnInlineConfig: true` in the options. `warnInlineConfig` is ignored as an unknown option.

Reply:
🤯 these options are such a mess.

Reply:
Okay, I'm clearly not understanding this. I've updated the tests to set `allowInlineConfig: true`, which I *think* should set `warnInlineConfig` to a truthy value inside of `Linter`, and the tests are still passing without any changes to `Linter`.

It would be helpful if you could let me know what I *should* be doing rather than what I shouldn't be doing.

Reply:
Linter was already fixed for the case these tests are targeting (maybe you mixed up this thread with https://github.com/eslint/eslint/pull/17351#discussion_r1327200617?). To avoid further confusion, I'll close all open threads, check everything again, and leave new comments on the latest code.

Reply:
Sounds good, thanks!